kevm-pyk/src/kevm_pyk/kproj/evm-semantics/lemmas/evm-int-simplification.k

```diff
@@ -52,4 +52,27 @@ module EVM-INT-SIMPLIFICATION-COMMON
       requires 0 <Int Y
       [simplification, comm, preserves-definedness]
 
+  // ###########################################################################
+  // #asWord
+  // ###########################################################################
+
+  rule [asWord-lt]:
+    #asWord ( BA ) <Int X => true
+    requires 2 ^Int ( lengthBytes(BA) *Int 8) <=Int X
+    [concrete(X), simplification]
+
+  rule [asWord-lt-range]:
+    #asWord ( #range ( _:Bytes, S, W ) ) <Int X => true
+    requires 0 <=Int S andBool 0 <=Int W
+     andBool X ==Int 2 ^Int log2Int( X )
+     andBool 2 ^Int ( 8 *Int W ) <=Int X
+    [simplification, concrete(S, W, X)]
```

Comment:
```suggestion
    requires 0 <=Int S andBool 0 <=Int W
     andBool 2 ^Int ( 8 *Int W ) <=Int X
     andBool X ==Int 2 ^Int log2Int( X )
    [simplification, concrete(S, W, X)]
```

This may behave better (depending on how the backend orders constraints when looking at them). But this will throw out the chance of `log2Int(X)` crashing if `X <=Int 0`, because the priory two constraints imply it must be greater than 0.

Reply:
I can change it throughout, yes, assuming constraints are processed in order. I've talked with @geo2a and, from what I remember, there is no guarantee on the processing order. I also remember trying this out and the LLVM backend still crashing with a 0.